sklearn_numba_dpex/kmeans/kernels/compute_labels.py

```diff
@@ -27,7 +27,7 @@ def make_label_assignment_fixed_window_kernel(
         work_group_size,
         device,
         required_local_memory_per_item=dtype_itemsize,
-        required_memory_constant=sub_group_size * dtype_itemsize,
+        required_memory_constant=window_n_centroids * dtype_itemsize,
     )
 
     centroids_window_width = window_n_centroids
```

Comment:
Maybe we could do:

```suggestion
    centroids_window_width = sub_group_size
```

and completely drop the `window_n_centroids` variable to remove on indirection?

Reply:
I propose removing `centroids_window_width` completely here. (see previous comment)

Now, regarding using `sub_group_size` or `window_n_centroids` I don't know what is better. In the current iteration I've favored using `window_n_centroids`.